graph-gateway/src/network/internal.rs

```diff
@@ -955,19 +881,74 @@ pub fn construct_network_topology_snapshot(
         })
         .collect::<HashMap<_, _>>();
 
-    // Construct the topology subgraphs table
+    // Construct the transferred subgraphs table
+    let transferred_subgraphs = subgraphs_info_table
+        .iter()
+        .filter_map(|(subgraph_id, subgraph)| {
+            // A subgraph is considered to be transferred to L2 if all its versions-deployments
+            // are transferred to L2 (i.e., `transferred_to_l2` is `true`) and have no allocations.
+            let transferred_to_l2 = subgraph.versions.iter().all(|version| {
+                version.deployment.transferred_to_l2 && version.deployment.allocations.is_empty()
+            });
+
+            // If the subgraph is transferred to L2 and has an ID on L2, return the pair.
+            // Otherwise, exclude the subgraph.
+            if transferred_to_l2 && subgraph.id_on_l2.is_some() {
+                Some((*subgraph_id, subgraph.id_on_l2?))
+            } else {
+                None
+            }
+        })
+        .collect::<HashMap<_, _>>();
+
+    // Construct the transferred deployments table
+    let transferred_deployments = deployments_info_table
+        .iter()
+        .filter_map(|(deployment_id, deployment)| {
+            if deployment.transferred_to_l2 && deployment.allocations.is_empty() {
+                Some(*deployment_id)
+            } else {
+                None
+            }
+        })
+        .collect::<HashSet<_>>();
+
+    // Construct the subgraphs table
     let subgraphs = subgraphs_info_table
         .into_iter()
         .filter_map(|(subgraph_id, subgraph)| {
-            let highest_version = subgraph.versions.first();
+            // Filter-out the subgraphs' invalid versions-deployments.
+            // If all the subgraph's versions are invalid, exclude the subgraph.
+            let versions: Vec1<_> = subgraph
+                .versions
+                .into_iter()
+                .filter(|version| {
+                    // Valid version must have a deployment with:
+                    // - Valid manifest info (network and start block).
+                    // - Not marked transferred to L2.
+                    // - At least one allocation.
+                    version.deployment.manifest_network.is_some()
+                        && version.deployment.manifest_start_block.is_some()
```

Comment:
I don't think the gateway has any reason to require a `startBlock`. We should default it to 0 and allow if None.